version: "3"
services:
  datascience-notebook:
    image: jupyter/datascience-notebook
    user: root
    working_dir: /home/madpin
    container_name: jupyter
    network_mode: bridge
    volumes:
      - ${HOME}/ipynbs:/home/madpin
    environment:
      # - PUID=1000
      # - PGID=1000
      - TZ=Europe/Dublin
      - JUPYTER_ENABLE_LAB=yes
      - RESTARTABLE=yes
      - NB_USER=madpin
      - CHOWN_HOME=yes
      # - LETSENCRYPT_HOST=jupyter.madpin.dev
      # - LETSENCRYPT_EMAIL=madpin@gmail.com
      - VIRTUAL_HOST=jupyter.madpin.dev
      - VIRTUAL_PORT=8888
    # ports:
    #   - 8888:8888
    restart: unless-stopped
    command: start-notebook.sh --NotebookApp.password='sha1:47a87a60d69d:e98216612acf96544b2ea28b49bd25db03b58197' --ip='*'
